pandas/core/sorting.py

```diff
@@ -239,7 +241,9 @@ def nargsort(items, kind='quicksort', ascending=True, na_position='last'):
     """
 
     # specially handle Categorical
-    if is_categorical_dtype(items):
+    if is_extension_array_dtype(items):
+        if isinstance(items, ABCIndexClass):
```

Comment:
Hmm, so when we have ExtensionArrays-backed indexes in the future, this is liable to break? Or we'll need a `._values` as part of the interface...

I don't suppose `items = items._values_for_argsort` works here? In theory, DatetimeArray could just use i8 values for sorting, bypassing the warning?

Reply:
`._values` is defined to be "the best array representation", so if the Index is backed by an EA, it will always return the EA? (also if we have an ExtensionIndex). I can maybe also use the public `.array` now?

> I don't suppose items = items._values_for_argsort

No, because we want to dispatch to the `argsort` method of the EA, I think?

Reply:
Sorry, I mis-remembered where the warning was coming from (I thought it was implemented on DatetimeArray).

Reply:
why exactly is this change needed? should this not just defer to the index ``.argsort``, which for DTA handles this?

Reply:
The `Index.argsort` has a different signature (and that's not specific to DatetimeTZ, but the general Index.argsort, so I would rather not deal with that here. Can open an issue for that)